src/derivative_utils.jl

```diff
@@ -521,7 +486,10 @@ function build_J_W(alg,u,uprev,p,t,dt,f,uEltypeNoUnits,::Val{true})
       return nothing, nothing
     end
   end
-  if ArrayInterface.isstructured(f.jac_prototype) || f.jac_prototype isa SparseMatrixCSC
+  n = length(u)
+  if f.jac_prototype !== nothing && size(f.jac_prototype) != (n, n) ||
```

Comment:
This is what we need to think more carefully about. Right now it's:

If it's some interesting array type, similar it
Otherwise if it has a Jacobian function on it and the jac_prototype is given, then it must be some weird array type so let's go with that operator
Otherwise it's the basic case.

But the middle case: do you need a Jacobian function to be a proper WOperator equation?